src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -488,16 +544,17 @@ def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
 
         context = dict(
             prompts=original_inputs,
-            num_generated_predictions=inputs.num_generated_predictions,
-            return_logits=inputs.return_logits,
+            num_return_sequences=num_return_sequences,
+            return_logits=generation_config.output_scores,
             include_prompt_logits=inputs.include_prompt_logits,
             callback=inputs.callback,
             stop=inputs.stop,
-            top_p=inputs.top_p,
-            top_k=inputs.top_k,
+            top_p=generation_config.top_p,
+            top_k=generation_config.top_k,
             presence_penalty=inputs.presence_penalty,
-            frequency_penalty=inputs.frequency_penalty,
-            max_tokens=inputs.max_tokens,
+            frequency_penalty=generation_config.repetition_penalty,
+            max_tokens=max_length,
+            max_new_tokens=generation_config.max_new_tokens,
```

Comment:
why not pass generation config in context directly instead of individually named values?